app/apps/handlers.js

```diff
@@ -1,12 +1,16 @@
 const { App, Bucket, User } = require('../models');
 const { Repo } = require('../models/github');
+const cls = require('continuation-local-storage');
 const config = require('../config');
-const github = require('../api_clients/github');
+const { GithubAPIClient } = require('../api_clients/github');
 const { url_for } = require('../routes');
 
 
 exports.new = (req, res, next) => {
-  github.api.authenticate(req.user)
+  const github_api = new GithubAPIClient(config);
+  const ns = cls.getNamespace(config.continuation_locals.namespace);
+  ns.set('github', github_api);
```

Comment:
Few questions: Where is the stuff set goes? Is there any limitation on how big the values set in the namespace are? Do you reckon serialisation of these objects would cause problems?

Reply:
The objects are not serialized - the namespace object is in memory for the duration of the request handling function call chain (eg: continuation middleware calls `next()` -> auth middleware calls `next()` -> route handler calls `res.send()`), then it is destroyed.

Reply:
I had a quick look at the module code and that's what I thought 👍